src/pages/index.js

```diff
@@ -140,39 +196,117 @@ const addCardPopup = new PopupWithForm(
 addCardPopup.setEventListeners();
 
 // Add Card Button
+const addCardButton = document.querySelector("#add-card-button");
+
 addCardButton.addEventListener("click", () => {
-  // Reset the Input fields on the add card popup
   addCardFormValidator.resetValidation();
 
   addCardPopup.open();
 });
 
+// Create a new card
 function handleCardSaveSubmit(inputValues) {
-  // Create a new card instance
-  const newCardInstance = renderCard({
+  const saveButton = document.getElementById("add-card-save-button");
+
+  saveButton.textContent = "Saving...";
+
+  const cardInfo = {
     name: inputValues.title,
     link: inputValues.url,
+  };
+
+  api
+    .addNewCard(cardInfo)
+    .then((res) => {
+      console.log("New Card:", cardInfo);
+      renderCard(res);
+    })
+    .catch((error) => {
+      console.error("API Error:", error);
+    })
+    .finally(() => {
+      saveButton.textContent = "Save";
+      addCardPopup.close();
+    });
+}
+
+// *************************** Delete Card  ************************** //
+const confirmDeletePopup = new PopupWithSubmit({
+  popupSelector: "#delete-card-modal",
+});
+
+const confirmDeleteButton = document.querySelector(
+  "#confirm-delete-modal-button"
+);
+
+confirmDeletePopup.setEventListeners();
+
+function handleDeleteIconClick(card) {
+  confirmDeletePopup.open(card);
+  confirmDeleteButton.focus();
+  confirmDeletePopup.setSubmit(() => {
+    card._deleteCard();
+    confirmDeletePopup.close();
   });
 }
 
-// ************************ Inital Card Setup ************************ //
+// ************************** Preview Image ************************** //
+
+// Preview Image Popup
+const previewImagePopup = new PopupWithImage({
+  popupSelector: "#preview-image-modal",
+});
+
+// Clicking an Image to get the preview Image
+function handleImageClick(cardData) {
+  previewImagePopup.open(cardData);
+}
+
+previewImagePopup.setEventListeners();
+
+const renderCard = (card) => {
+  const cardInstance = new Card(
+    card,
+    "#card-template",
+    handleImageClick,
+    handleDeleteIconClick,
+    api
+  );
+
+  const cardElement = cardInstance.generateCard();
+  cardSection.addItem(cardElement);
+};
 
+// ************************* Validation ************************* //
+const formSettings = {
```

Comment:
From the 8th sprint all constants (`initialCards` and `formSettings`) should be in file `constants.js` in folder `utils` to clean up the main file.